python/dgl/distributed/dist_graph.py

```diff
@@ -0,0 +1,506 @@
+
+    def __delitem__(self, key):
+        #TODO how to delete data in the kvstore.
+        pass
+
+    def __len__(self):
+        return self._graph.number_of_nodes()
+
+    def __iter__(self):
+        return iter(self._graph._get_all_ndata_names())
+
+    def __repr__(self):
+        name_list = self._graph._get_all_ndata_names()
+        reprs = {}
+        for name in name_list:
+            dtype, shape, _ = self._graph._client.get_data_meta(_get_ndata_name(name))
+            reprs[name] = 'DistTensor(shape={}, dtype={})'.format(str(shape), str(dtype))
+        return repr(reprs)
+
+class EdgeDataView(MutableMapping):
+    """The data view class when G.edges[...].data is called.
+
+    See Also
+    --------
+    dgl.DGLGraph.edges
+    """
+    __slots__ = ['_graph', '_graph_name']
+
+    def __init__(self, graph, graph_name):
+        self._graph = graph
+        self._graph_name = graph_name
+
+    def __getitem__(self, key):
+        return DistTensor(self._graph, _get_edata_name(key))
+
+    def __setitem__(self, key, val):
+        raise DGLError("DGL doesn't support set items. Please call init_ndata.")
+
+    def __delitem__(self, key):
+        #TODO
+        pass
+
+    def __len__(self):
+        return self._graph.number_of_edges()
+
+    def __iter__(self):
+        return iter(self._graph._get_all_edata_names())
+
+    def __repr__(self):
+        name_list = self._graph._get_all_edata_names()
+        reprs = {}
+        for name in name_list:
+            dtype, shape, _ = self._graph._client.get_data_meta(_get_edata_name(name))
+            reprs[name] = 'DistTensor(shape={}, dtype={})'.format(str(shape), str(dtype))
+        return repr(reprs)
+
+
+class DistGraphServer(KVServer):
+    ''' The DistGraph server.
+
+    This DistGraph server extends KVStore server. Its main function is to load the graph data
+    and share the data in its own partition with shared memory so that the DistGraph client
+    can access the data in the local partition with shared memory.
+
+    Parameters
+    ----------
+    server_id : int
+        KVServer's ID (start from 0).
+    server_namebook: dict
+        IP address namebook of KVServer, where key is the KVServer's ID
+        (start from 0) and value is the server's machine_id, IP address and port, e.g.,
+
+          {0:'[0, 172.31.40.143, 30050],
+           1:'[0, 172.31.40.143, 30051],
+           2:'[1, 172.31.36.140, 30050],
+           3:'[1, 172.31.36.140, 30051],
+           4:'[2, 172.31.47.147, 30050],
+           5:'[2, 172.31.47.147, 30051],
+           6:'[3, 172.31.30.180, 30050],
+           7:'[3, 172.31.30.180, 30051]}
+    num_client : int
+        Total number of client nodes.
+    graph_name : string
+        The name of the graph.
+    conf_file : string
+        The path of the config file generated by the partition tool.
+    '''
+    def __init__(self, server_id, server_namebook, num_client, graph_name, conf_file):
```

Comment:
Can we have a filepath instead of a graph_name? It is not clear to me why need this name and how it is used.